core/adapters/eth_tx.go

```diff
@@ -239,3 +247,46 @@ func pendingConfirmationsOrConnection(input models.RunInput) models.RunOutput {
 	}
 	return models.NewRunOutputPendingConnection()
 }
+
+// checkForShortCircuitedFormat checks whether etx has requested a format option
+// which short-circuits the usual processing of the output transaction bytes, in
+// order to give the user more control. defaultPrefix should concatenate its
+// inputs and prefix them appropriately (e.g., add the function selector and
+// initial offset into the arguments.)
+func checkForShortCircuitedFormat(
+	e *EthTx, result gjson.Result, defaultPrefix func(...[]byte) []byte) (
+	shortCircuited bool, output []byte, err error) {
+	switch e.DataFormat {
+	case utils.FormatRawHexWithFuncSelectorAndDataPrefix:
+		// TODO(alx): Should we enforce 0x-prefix, here? Might break existing jobs...
```

Comment:
I see a TODO here, should this PR be a draft?

Reply:
No, it's something I wanted addressed in review.